test/cbuilder_test.lua

```diff
@@ -0,0 +1,253 @@
+local t = require('luatest')
+
+local config_builder = require('luatest.cbuilder')
+local utils = require('luatest.utils')
+
+local DEFAULT_CONFIG = {
+    credentials = {
+        users = {
+            client = {password = 'secret', roles = {'super'}},
+            replicator = {password = 'secret', roles = {'replication'}},
+        },
+    },
+    iproto = {
+        advertise = {peer = {login = 'replicator'}},
+        listen = {{uri = 'unix/:./{{ instance_name }}.iproto'}},
+    },
+    replication = {timeout = 0.1},
+}
+
+local function merge_config(base, diff)
+    if type(base) ~= 'table' or type(diff) ~= 'table' then
+        return diff
+    end
+    local result = table.copy(base)
+    for k, v in pairs(diff) do
+        result[k] = merge_config(result[k], v)
+    end
+    return result
+end
+
+local g = t.group()
+
+g.test_default_config = function()
+    t.run_only_if(utils.version_current_ge_than(3, 0, 0),
```

Comment:
To run these tests on CI we should add 3.0 version to the workflow matrix. Please do it.

Reply:
Fixed. I also added `2.11` version and bump `setup-tarantool@v1` to `setup-tarantool@v3`:

```diff
diff --git a/.github/workflows/test_on_push.yaml b/.github/workflows/test_on_push.yaml
index fcabca1..3d5462c 100644
--- a/.github/workflows/test_on_push.yaml
+++ b/.github/workflows/test_on_push.yaml
@@ -11,12 +11,12 @@ jobs:
       github.event_name == 'pull_request' && github.event.pull_request.head.repo.owner.login != 'tarantool'
     strategy:
       matrix:
-        tarantool: ["1.10", "2.6", "2.7", "2.8", "2.10"]
+        tarantool: ["1.10", "2.6", "2.7", "2.8", "2.10", "2.11", "3.0"]
       fail-fast: false
     runs-on: [ubuntu-20.04]
     steps:
       - uses: actions/checkout@master
-      - uses: tarantool/setup-tarantool@v1
+      - uses: tarantool/setup-tarantool@v3
         with:
           tarantool-version: '${{ matrix.tarantool }}'
```